drivers/video/video_device.h

```diff
@@ -9,15 +9,19 @@
 
 #include <zephyr/device.h>
 
+struct video_ctrl_handler;
+
 struct video_device {
 	const struct device *dev;
 	const struct device *child_dev;
+	struct video_ctrl_handler *ctrl_handler;
 };
```

Comment:
A draft that tries to sketch the relations between devices here:

![zephyr_video_device_control_framework](https://github.com/user-attachments/assets/4b6ea60d-b56e-44c9-841f-d88c638ba20f)

If that is correct, it seems like it would be possible to reduce the implementation just a little bit by merging ctrl_handler in:

```suggestion
struct video_device {
	const struct device *dev;
	const struct device *child_dev;
	sys_slist_t ctrls;
};
```

Then the functions everywhere would use `struct video_device *vdev` instead of `struct video_ctrl_handler *hdl`.

By propagating the changes, this would merge `video_ctrl_handler_init()` with `video_device_find()`... Not huge advantage, but it also makes the model easier to grasp and learn.

![zephyr_video_device_control_framework_shrunk](https://github.com/user-attachments/assets/04a0fbd5-b7e4-414a-b913-8224b7b50b61)

Maybe there was more planned for that `struct video_ctrl_handler` though! Like having one ctrl_handler for two distinct devices...

Reply:
Yes, `struct video_ctrl_handler` is intended to hold more thing in the future but I don't have clear idea yet for now. So, it's better to keep things as simplest as possible for the current scope of the PR. It's the same as the discussion on the extension for multiple sensors / virtual channel which should be reserved for the future to not to complicate the scope of the PR.